sdk/storage/Azure.Storage.Blobs/src/AppendBlobClient.cs

```diff
@@ -330,6 +330,175 @@ protected sealed override BlobBaseClient WithCustomerProvidedKeyImpl(CustomerPro
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{BlobContentInfo}"/> describing the
+        /// newly created append blob.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="StorageRequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual async Task<Response<BlobContentInfo>> CreateIfNotExistsAsync(
+            BlobHttpHeaders? httpHeaders = default,
+            Metadata metadata = default,
+            AppendBlobAccessConditions? accessConditions = default,
+            CancellationToken cancellationToken = default) =>
+            await CreateIfNotExistsInternal(
+                httpHeaders,
+                metadata,
+                accessConditions ?? default,
+                true, // async
+                cancellationToken)
+                .ConfigureAwait(false);
+
+        /// <summary>
+        /// The <see cref="CreateIfNotExistsInternal"/> operation creates a new 0-length
+        /// append blob.  If the append blob already exists, the content of
+        /// the existing append blob will remain unchanged.  To add content to the append
+        /// blob, call the <see cref="AppendBlockAsync"/> operation.
+        ///
+        /// For more information, see <see href="https://docs.microsoft.com/rest/api/storageservices/put-blob" />.
+        /// </summary>
+        /// <param name="httpHeaders">
+        /// Optional standard HTTP header properties that can be set for the
+        /// new append blob.
+        /// </param>
+        /// <param name="metadata">
+        /// Optional custom metadata to set for this append blob.
+        /// </param>
+        /// <param name="accessConditions">
+        /// Optional <see cref="AppendBlobAccessConditions"/> to add
+        /// conditions on the creation of this new append blob.
+        /// </param>
+        /// <param name="async">
+        /// Whether to invoke the operation asynchronously.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{BlobContentInfo}"/> describing the
+        /// newly created append blob.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="StorageRequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        private async Task<Response<BlobContentInfo>> CreateIfNotExistsInternal(
+            BlobHttpHeaders? httpHeaders,
+            Metadata metadata,
+            AppendBlobAccessConditions accessConditions,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            if (accessConditions.HttpAccessConditions.HasValue)
```

Comment:
I kind of wonder if we should throw if `HttpAccessConditions` is set or maybe not even provide it on the `IfNotExists` overloads because anyone who wants to use this can just call the primary `Create` method.  Anyone else have opinions on this?

Reply:
I'm starting to lean towards throwing/not accepting HttpAccessConditions Because if the user sets the other headers, it could cause the service to throw different error codes depending on which headers they set and if the blob already exists or not. And in that case we might not catch the exception due to a different error code being thrown.

Reply:
I think that makes a lot of sense.  How about we just remove the `accessConditions` params from the `IfNotExists` helpers for this release?  We can always add them back later if we decide there are scenarios not served by having customers call `Create` directly.